src/lib/wallet/GoodWalletClass.js

```diff
@@ -440,18 +440,35 @@ export class GoodWallet {
 
   async syncTxFromExplorer(startBlock) {
     let results = []
-    const { tokenContract, account, networkId } = this
+    const { account, networkId, tokenContract, oneTimePaymentsContract } = this
     const { _address: tokenAddress } = tokenContract || {}
+    const { _address: otpAddress } = oneTimePaymentsContract || {}
+    log.info('polltest - syncTxFromExplorer', { startBlock, tokenAddress, account, networkId, otpAddress })
 
+    // this is both send/receive events
+    // missing otplEvents
+
+    const txResolve = []
     if (tokenAddress) {
-      results = await API.getTokenTxs(tokenAddress, account, networkId, startBlock).then(results =>
-        results.map(result => ({ ...result, transactionHash: result.hash })),
+      results = txResolve.push(
```

Comment:
why not push here also the logs query? so they happen in parallel

Reply:
ye that is goal, it was just where I left off on friday